### ISSUE TITLE
Question: Backup Block Volumes

### ISSUE THREAD

**Issue by catsem** [gilbertchen/duplicacy#345]
Hey,

is there a way to backup block devices with duplicacy, like lvm snapshots, partitions or disks?
If the deduplication works on block level, why only allow backups on file/directory level?

Best Regards
Chris

**Comment by gilbertchen** [comment]
Yes, the deduplication should work on block level.  Duplicacy supports the fixed-size chunking algorithm that is more suitable for deduplication at block level because of better deduplication ratio and fast speed (no rolling hash computation).  To enable that, you should set all 3 chunk size parameters to the same value when initializing a storage.  For instance:

```
duplicacy init -c 1M -max 1M -min 1M repository_id storage_url
```

These are the exact parameters we use for [Vertical Backup](https://www.verticalbackup.com), a special edition of Duplicacy made for VMWare ESXi to back up virtual machines.

**Comment by catsem** [comment]
The point is that `duplicacy init` only works for directorys, since you have to be in the target directory when you execute the init command. I cannot cd to /dev/mapper/myvg-mylv or /dev/sda2, so there seems to be no way to init block devices at the moment.

From what I can see on the website, vertical backup is very limited to vmware (which we do not use).

There seems to be a gap in both products or am I missing something?

**Comment by gilbertchen** [comment]
You can actually run `cd /dev && duplicacy init ...`.  The real issue is that Duplicacy doesn't read non-regular files.  If you could read `/dev/sda2` as a regular file then it would work.
